### PR TITLE
XRay Remote Sampler - Preserve previous rule reservoir if rule property has not changed

### PR DESCRIPTION
**Pull request by jj22ee**
Tracking issue: #3619
Issue summary: When Sampling Rules are updated, the previous rule reservoir is removed, which will take away any sampling quota from the user. When Rules are updated very frequently, the reservoir is removed frequently. The previous rule should be preserved if the rule property hasn't changed.

FIx:
When updating Rules, if the new `Rule.ruleProperties` hasn't changed, then preserve the previous Rule (which will also preserve the Rule reservoir and Rule samplingStatistics).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - AWS SDK span name to be of the format `Service.Operation` in `go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws`. (#3582, #3521)
 - Prevent sampler configuration reset from erroneously sampling first span in `go.opentelemetry.io/contrib/samplers/jaegerremote`. (#3603, #3604)
+- AWS XRay Remote Sampling to preserve previous rule if updated rule property has not changed in `go.opentelemetry.io/contrib/samplers/aws/xray`. (#3619, # TBD)
```
Comment (dmathieu) [inline CHANGELOG.md:20]: Changelog entries link to the PR making a change, not issues 😄 

```suggestion
- AWS XRay Remote Sampling to preserve previous rule if updated rule property has not changed in `go.opentelemetry.io/contrib/samplers/aws/xray`. (#3620)
```

**Comment by dmathieu** [comment]
Could you add tests?

**Review by willarmiros — APPROVED** [review]
lgtm!

File: CHANGELOG.md (hunk 1 above)
Comment (jj22ee) [reply CHANGELOG.md]: Updated with both issue and PR link

**Comment by jj22ee** [comment]
@Aneurysm9 Pinging for review!

**Review by Aneurysm9 — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Comment by jj22ee** [comment]
@Aneurysm9 @dmathieu
Can this be merged?

File: CHANGELOG.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - AWS SDK rename attributes `aws.operation`, `aws.service` to `rpc.method`,`rpc.service` in `go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws`. (#3582, #3617) 
 - AWS SDK span name to be of the format `Service.Operation` in `go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws`. (#3582, #3521)
 - Prevent sampler configuration reset from erroneously sampling first span in `go.opentelemetry.io/contrib/samplers/jaegerremote`. (#3603, #3604)
+- AWS XRay Remote Sampling to preserve previous rule if updated rule property has not changed in `go.opentelemetry.io/contrib/samplers/aws/xray`. (#3619, #3620)
```
[the remark this answers is not in the document]
Comment (jj22ee) [reply CHANGELOG.md]: Fixed.

**Pull request merged**